base_layer/core/src/proto/transaction.rs

```diff
@@ -202,10 +211,7 @@ impl TryFrom<TransactionInput> for proto::types::TransactionInput {
                     .to_vec(),
                 // Output hash is only used in compact form
                 output_hash: Vec::new(),
-                covenant: input
-                    .covenant()
-                    .map_err(|_| "Non-compact Transaction input should contain covenant".to_string())?
-                    .to_bytes(),
+                covenant,
```

Comment:
```suggestion
                input
￼                    .covenant()
￼                    .map_err(|_| "Non-compact Transaction input should contain covenant".to_string())?
￼                    .serialize()?,
```

Reply:
Output of the serialize is not the buffer, it's just `()`